base/deprecated.jl

```diff
@@ -1724,6 +1724,9 @@ end
 
 @deprecate atan2(y, x) atan(y, x)
 
+# PR #27432
+@eval @deprecate $(Symbol("@pure")) $(Symbol("@unsafe_pure")) false
```

Comment:
I think we can keep `@pure` as an (non-deprecated) alias for `@unsafe_pure` too, to avoid breaking the package ecosystem. In the future, we'll weaken the meaning of `@pure` (probably to something like `speculatable`), but preferably, most package uses should be able to continue using `@pure` though that transition. That'll also let me audit Base – as you conveniently offered :) – to indicate which expect to still need the unsafe version.